Source/CarthageKit/BuildSettings.swift

```diff
@@ -44,12 +44,8 @@ public struct BuildSettings {
 	/// Upon .success, sends one BuildSettings value for each target included in
 	/// the referenced scheme.
 	public static func load(with arguments: BuildArguments, for action: BuildArguments.Action? = nil) -> SignalProducer<BuildSettings, CarthageError> {
-		// xcodebuild (in Xcode 8) has a bug where xcodebuild -showBuildSettings
-		// can hang indefinitely on projects that contain core data models.
-		// rdar://27052195
-		// Including the action "clean" works around this issue, which is further
-		// discussed here: https://forums.developer.apple.com/thread/50372
-		let task = xcodebuildTask(["clean", "-showBuildSettings", "-skipUnavailableActions"], arguments)
+		// Use "archive" action here to determine if it's available to build
+		let task = xcodebuildTask(["archive", "-showBuildSettings", "-skipUnavailableActions"], arguments)
```

Comment:
Does this still work around rdar://27052195 correctly?

Reply:
Aha! That's Xcode 8 issue.. I'm now installing Xcode 8 to test it, so please hold on.

Reply:
The issue is originally reported in Xcode 8, but we are not sure that is fixed in Xcode 9 or later. We don't explicitly drop building frameworks with Xcode 8, so we should keep the workaround as much as possible.

Reply:
Hmm, I could not reproduce this issue.. Maybe my machine is too powerful (iMac Pro🤪)

Reply:
I could reproduce this issue and using `archive` here also fix the issue on Xcode 8.0. I'll put the note here soon.